File: app/src/main/java/com/mpcorporation/snapeffect/Adapter/EffectAdapter.java
package com.mpcorporation.snapeffect.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mpcorporation.snapeffect.Model.EffectItem;
import com.mpcorporation.snapeffect.R;

import java.util.List;

import jp.co.cyberagent.android.gpuimage.filter.GPUImageFilter;

public class EffectAdapter extends RecyclerView.Adapter<EffectAdapter.ViewHolder> {

    public interface OnEffectClickListener {
        void onClick(GPUImageFilter filter);
    }

    private final List<EffectItem> effects;
    private final OnEffectClickListener listener;

    public EffectAdapter(List<EffectItem> effects, OnEffectClickListener listener) {
        this.effects = effects;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.filter_component, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        EffectItem item = effects.get(position);
        holder.textName.setText(item.getName());
        holder.imageIcon.setImageResource(item.getFilterIconRes());
        holder.itemView.setOnClickListener(v -> listener.onClick(item.getFilter()));
    }

    @Override
    public int getItemCount() {
        return effects.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imageIcon;
        TextView textName;

        public ViewHolder(View view) {
            super(view);
            imageIcon = view.findViewById(R.id.filter_icon);
            textName = view.findViewById(R.id.filter_name);
        }
    }
}

File: app/src/main/java/com/mpcorporation/snapeffect/presentation/common/adapter/EffectAdapter.kt
package com.mpcorporation.snapeffect.presentation.common.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mpcorporation.snapeffect.R
import com.mpcorporation.snapeffect.domain.model.EffectItem

class EffectAdapter(
    private val effects: List<EffectItem>,
    private val onClick: (EffectItem) -> Unit
) : RecyclerView.Adapter<EffectAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val icon: ImageView = view.findViewById(R.id.filter_icon)
        val label: TextView = view.findViewById(R.id.filter_name)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.filter_component, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = effects[position]
        holder.label.text = item.name
        holder.icon.setImageResource(item.filterIconRes)
        holder.itemView.setOnClickListener { onClick(item) }
    }

    override fun getItemCount(): Int = effects.size
}
